pallets/automation-time/src/autocompounding.rs

```diff
@@ -0,0 +1,221 @@
+// This file is part of OAK Blockchain.
+
+// Copyright (C) 2022 OAK Network
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use sp_std::{vec, vec::Vec};
+
+#[derive(Clone)]
+pub struct IntervalRow {
+	interval_earnings_after_fee: i128,
+	stake: i128,
+	ownership: f64,
+}
+
+pub fn do_calculate_optimal_autostaking(
```

Comment:
if i'm reading this function correctly, i _think_ what's happening is:
1. the inner loop calculates the APY you earn with the various frequencies at which you auto compound, given a set duration you want to hold for. you store these APY values into a table.
2. the outer loop tries different durations and feeds those durations into the inner loop to calculate a "best" autocompounding frequency
3. you save the best of the best to get the best duration and frequency

is this the right interpretation?